airbyte-integrations/connectors/source-mysql/src/test-integration/java/io/airbyte/integrations/io/airbyte/integration_tests/sources/AbstractMySqlSourceDatatypeTest.java

```diff
@@ -279,7 +279,7 @@ protected void initTests() {
               .fullSourceDataType(fullSourceType)
               .airbyteType(JsonSchemaType.STRING_TIME_WITHOUT_TIMEZONE)
               // JDBC driver can process only "clock"(00:00:00-23:59:59) values.
-              .addInsertValues("'-22:59:59'", "'23:59:59'", "'00:00:00'")
```

Comment:
Why this change?

Reply:
I'm a bit confused why it passed before with a negative date. I'm having: 
`SQLState: S1009, Message: The value '-22:59:59' is an invalid TIME value. JDBC Time objects represent a wall-clock time and not a duration as MySQL treats them. If you are treating this type as a duration, consider retrieving this value as a string and dealing with it according to your requirements.`
`-22:59:59` looks as invalid value for me or should mysql handle it? Could you explain?

Reply:
I second what Subodh is saying here. The functional change in this PR does not seem related to this test change. However, if this test passes before the functional changes and does not pass after - something is wrong with the functional changes.

Reply:
It seems like a bug for me in mysql-connector-java checked (8.0.22 and 8.0.30). 

So if I use `useCursorFetch=true` mysql-connector-java will use `MysqlBinaryValueDecoder` for `MysqlType.FIELD_TYPE_TIME ` which will remove negative value by: 
```
        if (negative) {
            days *= -1;
        } 
```
But if I useCursorFetch=false Decoder will be `MysqlTextValueDecoder` which also has decodeTime method but it did change this transformation twice. First from negative to positive value and then back and in the end it checks if value is negative then throw an Error 🤔 

So what options do we have? 

1. Return useCursorFetch=true in this case mysql will create some tmp tables for selects queries. I did not notice the performance change with useCursorFetch true or false and setFetchSize(Integer.MIN_VALUE). Also open new ticket about removing useCursorFetch
2. Remove test with negative time as for now. But it can leads to issues if someone has a negative time value in db 
3. Try older mysql-connector-java but then we can face some other old issues. 

Not sure what else I can try as for now. Any advice? 

@grishick @subodh1810

Reply:
I pushed [changes](https://github.com/airbytehq/airbyte/pull/17236/commits/ca12f87512b44c2de2a78fc245061aa61fc26b9d). Please have a look 
@grishick @subodh1810

Reply:
Thank you for digging into this. My preference is to Return `useCursorFetch=true ` since you have not seen this affecting performance negatively while using  `setFetchSize(Integer.MIN_VALUE)`. Our test coverage is not exhaustive and so I am more concerned about what other behaviors may change because of switching to `MysqlTextValueDecoder` instead of `MysqlBinaryValueDecoder`.

Reply:
@grishick returning useCursorFetch=true is also fine for me since I did not notice performance change or DB resource consuming. I added it back  https://github.com/airbytehq/airbyte/pull/17236/commits/8257491e7d6a2ec912bb6877ee067a4a1afac17d

Reply:
I know this is an old thread but this is the most relevant place I could think of mentioning:
I'm running into an issue with a MySQL TIME type, with the following error:

```
SQLState: S1009, Message: The value '30:00:00' is an invalid TIME value. JDBC Time objects represent a wall-clock time and not a duration as MySQL treats them. If you are treating this type as a duration, consider retrieving this value as a string and dealing with it according to your requirements.
```

This is regardless of `useCursorFetch=true` or `useCursorFetch=false`, happens for both.
I'm currently debugging the issue, but any help or thoughts is most welcome.